src/read.rs

```diff
@@ -366,6 +379,35 @@ impl<R> WavReader<R> where R: io::Read {
         Ok(spec_ex)
     }
 
+    fn read_wave_format_ieee_float(mut reader: R, chunk_len: u32, spec: WavSpec)
+                                   -> Result<WavSpecEx> {
+        // When there is a PCMWAVEFORMAT struct, the chunk is 16 bytes long.
+        // The WAVEFORMATEX structs includes two extra bytes, `cbSize`.
+        let is_wave_format_ex = chunk_len == 18;
+
+        if !is_wave_format_ex && chunk_len != 16 {
+            return Err(Error::FormatError("unexpected fmt chunk size"));
+        }
+
+        if is_wave_format_ex {
+            // For WAVE_FORMAT_IEEE_FLOAT which we are reading, there should
+            // be no extra data, so `cbSize` should be 0.
+            let cb_size = try!(reader.read_le_u16());
+            if cb_size != 0 {
+                return Err(Error::FormatError("unexpected WAVEFORMATEX size"));
+            }
+        }
+
```

Comment:
Should there be a check here that verifies that `spec.bits_per_sample == 32`? Apparently 64 bits is also valid, but let’s not bother until somebody actually encounters a 64-bit float wav file in the wild.